macros/models/_sector/cex/cex_deposit_addresses.sql

```diff
@@ -0,0 +1,95 @@
+{% macro cex_deposit_addresses(blockchain, transactions, traces, erc20_transfers, cex_addresses) %}
+
+WITH first_txs AS (
+    SELECT block_time
+    , block_number
+    , hash AS tx_hash
+    , "from" AS tx_from
+    FROM {{ transactions }}
+    WHERE nonce=0
+    {% if is_incremental() %}
+    AND block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+    )
+
+, commingle_transactions AS (
+    SELECT txs.block_time
+    , txs.block_number
+    , "from" AS deposit_address
+    , to AS cex_address
+    , cex.cex_name
+    , cex.distinct_name
+    , txs.tx_hash
+    , 'native' AS deposit_token_type
+    FROM first_txs txs
+    INNER JOIN {{ traces }} traces ON txs.block_number=traces.block_number
+        AND txs.tx_hash=traces.tx_hash
+        AND (traces.call_type NOT IN ('delegatecall', 'callcode', 'staticcall') OR traces.call_type IS NULL)
+        AND traces.success
+        {% if is_incremental() %}
+        AND traces.block_time >= date_trunc('day', now() - interval '7' day)
+        {% endif %}
+    INNER JOIN {{ cex_addresses }} cex ON cex.address=traces.to
+    
+    UNION ALL
+    
+    SELECT t.evt_block_time AS block_time
+    , t.evt_block_number AS block_number
+    , t."from" AS deposit_address
+    , t.to AS cex_address
+    , cex.cex_name
+    , cex.distinct_name
+    , t.evt_tx_hash AS tx_hash
+    , 'erc20' AS deposit_token_type
+    FROM first_txs txs
+    INNER JOIN {{ erc20_transfers }} t ON txs.block_number=t.evt_block_number
+        AND txs.tx_hash=t.evt_tx_hash
+        {% if is_incremental() %}
+        AND t.evt_block_time >= date_trunc('day', now() - interval '7' day)
+        {% endif %}
+    INNER JOIN {{ erc20_transfers }} t2 ON txs.block_number=t2.evt_block_number
+        AND txs.tx_hash=t2.evt_tx_hash
+        AND t."from"=t2.to
+        AND t2.evt_block_time BETWEEN t.evt_block_time - interval '1' day AND t.evt_block_time
```

Comment:
why is there a range here?

Reply:
This is to check that if the deposit address deposited erc20s into global cex wallets, it was also funded in erc20 in the day before that. That's an extra filter I use to ensure its a deposit address, 1 day seems like a decent loockback window but I might need to tweak or even remove